Sources/Purchasing/ProductsManager.swift

```diff
@@ -126,6 +135,25 @@ private extension ProductsManager {
         return self.productsFetcherSK1.products(withIdentifiers: identifiers, completion: completion)
     }
 
+    func trackProductsRequestIfNeeded(_ startTime: Date,
+                                     storeKitVersion: StoreKitVersion,
+                                     error: PurchasesError?) {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *),
+           let diagnosticsTracker = self.diagnosticsTracker {
+            let responseTime = self.dateProvider.now().timeIntervalSince(startTime)
+            let errorMessage = (error?.userInfo[NSUnderlyingErrorKey] as? Error)?.localizedDescription
```

Comment:
Does this error contain more information if casted to `SKError` or `StoreKitError`? I have been doing this in #4253 

```
if let skError = error?.userInfo[NSUnderlyingErrorKey] as? SKError {
    storeKitErrorDescription = skError.code.trackingDescription
} else if let storeKitError = error?.userInfo[NSUnderlyingErrorKey] as? StoreKitError {
    storeKitErrorDescription = storeKitError.trackingDescription
} else {
    storeKitErrorDescription = nil
}
```

Reply:
Hmm that's a good question... Will try that out, thanks!

Reply:
Do we want to add the storekit error `trackingDescription` now that we have that available?

Reply:
Isn't that handled by the [extractStoreKitErrorDescription](https://github.com/RevenueCat/purchases-ios/pull/4247/files#diff-b080ecaed1fab9b6af95a6a40653aa15720509e51c3faa1aa71263f99a4ff1cbR147) a few lines below?

Reply:
oh yes, my bad. I looked for `trackingDescription` 😮‍💨